pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -148,16 +151,21 @@ func (p *Provider) createAMIOptions(ctx context.Context, nodeTemplate *v1alpha1.
 	if len(securityGroupsIDs) == 0 {
 		return nil, fmt.Errorf("no security groups exist given constraints")
 	}
+	associatePublicIpv4Addrs, err := p.subnetProvider.AssociatePublicIpv4Addrs(ctx, nodeTemplate)
```

Comment:
Should we just pass:
1. `true` if all are `true`
2. `false` if all are `false`
3. `nil` if there is a mix

This would give us nice parity among all the return conditions

Reply:
Discussed with @bwagner5 offline. We are going with the way that he has proposed. We'll do a check and return `false` if _all_ of the subnets do not have `AssociatePublicIpAddress` set, otherwise, it will be `nil` and we will defer to the subnet setting